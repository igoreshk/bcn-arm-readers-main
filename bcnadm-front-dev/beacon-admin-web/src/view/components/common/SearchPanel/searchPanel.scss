.searchPanelContainer {
  display: flex;
  align-items: center;
  min-width: 150px;
  margin-right: 24px;
  border: 0.5mm solid rgba(222, 222, 222, 0.8);
  border-radius: 50px;

  .icon {
    height: 30px;
    width: 30px;
    fill: #999999;
    margin-left: 10px;
  }

  .searchPanel {
    margin: 0 10px;
  }
}