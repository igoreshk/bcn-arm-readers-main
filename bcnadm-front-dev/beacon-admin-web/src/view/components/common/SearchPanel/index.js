import { WithSearchContext } from 'src/view/components/common/WithSearch';
import React from 'react';
import PropTypes from 'prop-types';
import Search from '@material-ui/icons/Search';
import { TextField } from '@material-ui/core';
import './searchPanel.scss';

export default function SearchPanel(props) {
  SearchPanel.propTypes = {
    containerClassName: PropTypes.string,
    name: PropTypes.string,
    searchClassName: PropTypes.string,
    textFieldClassName: PropTypes.string,
    textFieldRest: PropTypes.object
  };

  SearchPanel.defaultProps = {
    containerClassName: 'searchPanelContainer',
    name: 'search-panel-filter',
    searchClassName: 'icon',
    textFieldClassName: 'searchPanel'
  };

  return (
    <WithSearchContext.Consumer>
      {({ filter, setFilter }) => (
        <div className={`${props.containerClassName} ${props.className}`}>
          <Search className={props.searchClassName} />
          <TextField
            name={props.name}
            value={filter}
            onChange={setFilter}
            className={props.textFieldClassName}
            {...props.textFieldRest}
          />
        </div>
      )}
    </WithSearchContext.Consumer>
  );
}
