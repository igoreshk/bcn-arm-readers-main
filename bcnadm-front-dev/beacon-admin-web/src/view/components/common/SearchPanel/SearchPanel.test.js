import React from 'react';
import { mount, shallow } from 'enzyme';
import SearchPanel from './index';

const props = {
  onChange: jest.fn(),
  value: ''
};

const wrapper = mount(<SearchPanel {...props} />);
const shallowWrapper = shallow(<SearchPanel {...props}/>);
describe('WithSearch', () => {
  it('Should checks that props are passed properly', () =>
    expect(wrapper.props()).toEqual({
      ...props
    }));

  it('Should match snapshot', () => expect(shallowWrapper).toMatchSnapshot());
});
