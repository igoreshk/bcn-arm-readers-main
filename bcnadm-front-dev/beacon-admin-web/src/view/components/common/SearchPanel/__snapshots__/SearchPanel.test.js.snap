// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithSearch Should match snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
